cvat-core/src/lambda-manager.ts

```diff
@@ -73,32 +92,59 @@ class LambdaManager {
             task: taskID,
         };
 
-        const result = await serverProxy.lambda.call(model.id, body);
+        let result;
+        if (model.provider === ModelProviders.CVAT) {
+            result = await serverProxy.lambda.call(model.id, body);
+        } else {
+            result = await serverProxy.functions.call(model.id, body);
+        }
         return result;
     }
 
     async requests() {
-        const result = await serverProxy.lambda.requests();
+        const lambdaRequests = await serverProxy.lambda.requests();
+        const functionsRequests = await serverProxy.functions.requests();
+        const result = [...lambdaRequests, ...functionsRequests];
         return result.filter((request) => ['queued', 'started'].includes(request.status));
     }
 
-    async cancel(requestID): Promise<void> {
+    async cancel(requestID, functionID): Promise<void> {
         if (typeof requestID !== 'string') {
             throw new ArgumentError(`Request id argument is required to be a string. But got ${requestID}`);
         }
+        const model = this.cachedList.find((_model) => _model.id === functionID);
+        if (!model) {
+            throw new ArgumentError('Incorrect Function Id provided');
+        }
 
         if (this.listening[requestID]) {
             clearTimeout(this.listening[requestID].timeout);
             delete this.listening[requestID];
         }
-        await serverProxy.lambda.cancel(requestID);
+
+        const { provider } = model;
+        if (provider === ModelProviders.CVAT) {
+            await serverProxy.lambda.cancel(requestID);
+        } else {
+            await serverProxy.functions.cancel(requestID);
+        }
     }
 
-    async listen(requestID, onUpdate): Promise<void> {
+    async listen(requestID, functionID, onUpdate): Promise<void> {
+        const model = this.cachedList.find((_model) => _model.id === functionID);
+        if (!model) {
+            throw new ArgumentError('Incorrect Function Id provided');
+        }
+        const { provider } = model;
         const timeoutCallback = async (): Promise<void> => {
             try {
                 this.listening[requestID].timeout = null;
-                const response = await serverProxy.lambda.status(requestID);
+                let response = null;
+                if (provider === ModelProviders.CVAT) {
+                    response = await serverProxy.lambda.status(requestID);
+                } else {
+                    response = await serverProxy.functions.status(requestID);
+                }
```

Comment:
Can we reduce code dublication for this logic? I found 4 places.
For example we could return ``serverProxy.lambda`` or ``serverProxy.functions`` depending on ``provider`` value